examples/hpc/dla-driver/src/tensor3.rs

```diff
@@ -260,9 +260,8 @@ pub fn rescale(
         };
 
         channel_slice.map_inplace(|x| {
-            let value = (((*x as f32 * pre_scale) - output_zero as f32) / input_scale) * scale
-                + input_zero as f32;
-            *x = value.clamp(i8::MIN as f32, i8::MAX as f32) as i8
+             let value = (input_scale / scale) * (*x as f32 * pre_scale - input_zero as f32)
+                 + output_zero as f32;
         });
```

Comment:
Assignment missing from `map_inplace`. I think this line is a no-op as it is. Strange you're not getting any complaints from the compiler or clippy :)

Reply:
Yeah I guess `map_inplace` doesn't give much in the way of static analysis. I think almost an equivalent but safer alternative would be to use `into_iterator` to consume the original allocation. This will likely result in the same assembly code but also not allow such issues as possibly introduced by this diff.

Reply:
Returned this to its original state, as the assignment was accidentally removed. The new codegen only hits this part of the code in an edge so it didn't cause problems during my testing.